src/cdk/stack.ts

```diff
@@ -106,93 +28,149 @@ interface MiraStackProps {
   [x: string]: unknown
 }
 
-/**
- * Note that in Mira, a "stack" is always nested within a service context to
- * support CICD out of the box.
- */
-export class MiraStack extends NestedStack implements ExportOutputs {
-    initialized: Promise<unknown>;
-    static stackInstances: Record<string, Stack[]>;
-    parent: Construct
-    name: string
-    props: MiraStackProps
-
-    constructor (parent: Construct, name?: string, props?: MiraStackProps) {
-      if (!name) {
-        name = 'DefaultStack'
-      }
-      if (!MiraStack.stackInstances[name]) {
-        MiraStack.stackInstances[name] = []
-      }
-      const id = MiraStack.stackInstances[name].length
-
-      super(parent, `${name}-${id}`)
-      this.parent = parent
-      this.name = name
-      this.props = props || {}
-      MiraStack.stackInstances[name].push(this)
-      this.initialized = new Promise((resolve) => {
-        setTimeout(async () => {
-          await this.initialize()
-          resolve()
-        }, 1)
-      })
+export class MiraStack extends MiraObject {
+  static topLevelStacks: LooseObject
+  parent?: MiraStack
+  stack: cdk.Stack
+  props: MiraStackProps
+  constructor(name?: string, parent?: MiraStack, existingStack?: cdk.Stack, props?: MiraStackProps) {
```

Comment:
Why don't invert `name` and `parent`? Is a common pattern in CDK to have the first parameter as the parent.
The name `MiraStack` make me thinks that is a Stack, even if it isn't. Then as a developer I would like to have a similar signature. Though?

Reply:
Ah, really agree.  I will implement this and update here.